src/foam/parse/QueryParser.js

```diff
@@ -506,21 +506,36 @@ foam.CLASS({
       var query = this.grammar_.parseString(str, opt_name);
       return query && query.partialEval ? query.partialEval() : query;
     },
-    // Adapt binary predicate values.
-    function bin_(opts) {
-      if ( ! ( opts.arg1 && opts.arg1.adapt && opts.hasOwnProperty('arg2') ) )
-        return opts;
-
-      if ( Array.isArray(opts.arg2) ) {
-        var arg2 = opts.arg2;
-        for ( var i = 0; i < arg2.length; i++ ) {
-          arg2[i] = opts.arg1.adapt(undefined, opts.arg2[i]);
+    {
+      name: 'bin_',
+      documentation: `For binary ops like EQ(prop, value), the QueryParser may
```

Comment:
We build the parser dynamically by walking the Property axioms of the Model. At that point we know the type and specific parser for the property, so we could just build a type-sound parser.